core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/CoreRemoteMongoCollectionWriteModelContainer.java

```diff
@@ -44,21 +46,33 @@ public CoreRemoteMongoCollectionWriteModelContainer(
    * Commits the writes to the remote collection.
    */
   @Override
-  public void commit() {
+  public boolean commit() {
     final CoreRemoteMongoCollection<DocumentT> collection = getCollection();
     final List<WriteModel<DocumentT>> writeModels = getBulkWriteModels();
 
+    // define success as any one operation succeeding for now
```

Comment:
[q] why not make it based on all three operations succeeding?

Reply:
it's actually something we'll have to fix long term with the "container" logic Jason added, which is that previously when a single operation failed, we knew immediately and wouldn't get to the point where we emitted the event. Now, we can't tell individually what passes and what fails. This is problematic. I'd rather over-emit duplicated events than fail to emit a unique event.

Reply:
That seems arbitrary– there are plenty of workarounds for this, including returning multiple values here.

Reply:
it's a limitation in the result provided by Mongo